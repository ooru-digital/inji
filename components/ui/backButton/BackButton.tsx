import React from 'react';
import {I18nManager, TouchableOpacity} from 'react-native';
import {Icon} from 'react-native-elements';
import testIDProps from '../../../shared/commonUtil';
import {Theme} from '../styleUtils';

export const BackButton: React.FC<BackButtonProps> = (
  props: BackButtonProps,
) => {
  let containerStyle: object = Theme.Styles.backArrowContainer;
  if (props.customIconStyle)
    containerStyle = {...containerStyle, ...props.customIconStyle};
  return (
    <TouchableOpacity
      onPress={props.onPress}
      {...testIDProps('goBack')}
      style={{zIndex: 1}}>
      <Icon
        {...testIDProps('arrow-left')}
        name={I18nManager.isRTL ? 'arrow-right' : 'arrow-left'}
        type="material-community"
        onPress={props.onPress}
        containerStyle={containerStyle}
        color={Theme.Colors.Icon}
      />
    </TouchableOpacity>
  );
};

interface BackButtonProps {
  onPress: () => void;
  customIconStyle?: object;
}
