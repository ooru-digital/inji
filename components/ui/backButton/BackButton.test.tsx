import React from 'react';
import renderer from 'react-test-renderer';
import {BackButton} from './BackButton';

describe('Back button test', () => {
  it('should render the back button ', () => {
    const backButtonTree = renderer.create(<BackButton onPress={jest.fn} />);

    expect(backButtonTree).toMatchSnapshot();
  });
});
