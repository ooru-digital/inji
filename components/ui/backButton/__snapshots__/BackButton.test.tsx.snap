// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Back button test should render the back button  1`] = `
<View
  accessibilityLabel="goBack"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
      "zIndex": 1,
    }
  }
/>
`;
